cmd/src/server.rs

```diff
@@ -90,66 +92,82 @@ pub fn run_tikv(config: TiKvConfig) {
     // Do some prepare works before start.
     pre_start();
 
-    let mut tikv = TiKVServer::init(config);
-
     let _m = Monitor::default();
 
-    tikv.check_conflict_addr();
-    tikv.init_fs();
-    tikv.init_yatp();
-    tikv.init_encryption();
-    tikv.init_engines();
-    let gc_worker = tikv.init_gc_worker();
-    let server_config = tikv.init_servers(&gc_worker);
-    tikv.register_services();
-    tikv.init_metrics_flusher();
-    tikv.run_server(server_config);
-    tikv.run_status_server();
-
-    signal_handler::wait_for_signal(Some(tikv.engines.take().unwrap().engines));
-
-    tikv.stop();
+    if !config.raft_engine.enable {
+        let mut tikv = TiKVServer::<RocksEngine>::init(config);
+        tikv.check_conflict_addr();
+        tikv.init_fs();
+        tikv.init_yatp();
+        tikv.init_encryption();
+        tikv.init_engines();
+        let gc_worker = tikv.init_gc_worker();
+        let server_config = tikv.init_servers(&gc_worker);
+        tikv.register_services();
+        tikv.init_metrics_flusher();
+        tikv.run_server(server_config);
+        tikv.run_status_server();
+
+        signal_handler::wait_for_signal(Some(tikv.engines.take().unwrap().engines));
+        tikv.stop();
+    } else {
+        let mut tikv = TiKVServer::<RaftLogEngine>::init(config);
+        tikv.check_conflict_addr();
+        tikv.init_fs();
+        tikv.init_yatp();
+        tikv.init_encryption();
+        tikv.init_engines();
+        let gc_worker = tikv.init_gc_worker();
+        let server_config = tikv.init_servers(&gc_worker);
+        tikv.register_services();
+        tikv.init_metrics_flusher();
+        tikv.run_server(server_config);
+        tikv.run_status_server();
+
+        signal_handler::wait_for_signal(Some(tikv.engines.take().unwrap().engines));
+        tikv.stop();
+    }
```

Comment:
There are too many duplicate parts. It is better to use `Either` or wrap it with enum